src/query/set_query.rs

```diff
@@ -0,0 +1,201 @@
+use tantivy_fst::raw::CompiledAddr;
+use tantivy_fst::{Automaton, Map};
+
+use crate::query::{AutomatonWeight, Query, Weight};
+use crate::schema::Field;
+use crate::{Searcher, Term};
+
+/// A Term Set Query matches all of the documents containing any of the Term provided
+///
+/// Terms not using the right Field are discared.
+#[derive(Debug, Clone)]
+pub struct TermSetQuery {
+    field: Field,
+    terms: Vec<Term>,
+}
+
+impl TermSetQuery {
+    /// Create a Term Set Query
+    pub fn new<T: IntoIterator<Item = Term>>(field: Field, terms: T) -> Self {
```

Comment:
The API is unclear here.
`field` is redundant in the sense that terms contain a field too...
You did not document what happens if the terms are targetting different field either.

Can you at the very least document this?